src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -1934,9 +1934,36 @@ private class OffsetAndRule {
 
 
 #if PLATFORM_UNIX
-        static public TimeZoneInfo FindSystemTimeZoneById(string id) {
-            // UNIXTODO
-            throw new NotImplementedException();
+        static public TimeZoneInfo FindSystemTimeZoneById(string id)
+        {
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new ArgumentNullException("id");
+            }
+
+            // Special case for Utc since there is code that expects only 1 instance of the UTC time zone info to be created.
+            // see CachedData.GetCorrespondingKind()
+            if (String.Compare(id, c_utcId, StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                return TimeZoneInfo.Utc;
+            }
+
+            // UNIXTODO: use CachedData
+
+            string timeFilePath = Path.Combine(c_timeZoneDirectory, id);
+            if (!File.Exists(timeFilePath))
```

Comment:
The check in windows is case-insensitive.  I was just following it's lead.  That way callers can say "UTC" or "Utc" and they still get the same object.

From the Windows code:
```C#
        static public TimeZoneInfo FindSystemTimeZoneById(string id) {

            // Special case for Utc as it will not exist in the dictionary with the rest
            // of the system time zones.  There is no need to do this check for Local.Id
            // since Local is a real time zone that exists in the dictionary cache
            if (String.Compare(id, c_utcId, StringComparison.OrdinalIgnoreCase) == 0) {
                return TimeZoneInfo.Utc;
            }
```